include/slate/func.hh

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2023, University of Tennessee. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+// This program is free software: you can redistribute it and/or modify it under
+// the terms of the BSD 3-Clause license. See the accompanying LICENSE file.
+
+#ifndef SLATE_FUNC_HH
+#define SLATE_FUNC_HH
+
+#include "slate/enums.hh"
+
+#include <functional>
+
+namespace slate {
+
+//------------------------------------------------------------------------------
+/// @namespace slate::func
+/// A set of functions useful in SLATE's "lambda" constructors
+///
+namespace func {
+
+using ij_tuple = std::tuple<int64_t, int64_t>;
+
+//------------------------------------------------------------------------------
+// Block size functions
+
+//------------------------------------------------------------------------------
+/// Creates a uniform blocksize
+///
+/// @param[in] n
+///     Global matrix size.  Needed to correctly handle the last block size
+///
+/// @param[in] nb
+///     Block size
+///
+/// @retval The requested blocksize function
```

Comment:
This should be `@return`, here and elsewhere. With `@retval`, I think Doxygen will tag the return value as "The", with a description of "requested blocksize function". In the LU_info PR #111, I switched everything to using `@return` because `@retval` doesn't handle multi-word return values like `i < 0` and `i > 0` nicely.